resources/timerbar.ts

```diff
@@ -1,91 +1,135 @@
 export default class TimerBar extends HTMLElement {
-  static get observedAttributes() {
+  rootElement: HTMLElement;
+  foregroundElement: HTMLElement;
+  backgroundElement: HTMLElement;
+  leftTextElement: HTMLElement;
+  centerTextElement: HTMLElement;
+  rightTextElement: HTMLElement;
+  private _duration: number;
+  private _start: number;
+  private _width: string;
+  private _height: string;
+  private _bg: string;
+  private _fg: string;
+  private _towardRight: boolean;
+  private _styleFill: boolean;
+  private _leftText: string;
+  private _centerText: string;
+  private _rightText: string;
+  private _hideAfter: number;
+  private _loop: boolean;
+  private _connected: boolean;
+  private _hideTimer: number | null;
+  private _animationFrame: number | null;
+
+  static get observedAttributes(): string[] {
```

Comment:
I think at the very least, `styleFill` needs to be listed here.  That's not enough to fix the issue though, so something else needs to change as well.